llm/server.go

```diff
@@ -145,11 +145,11 @@ func LoadModel(model string, maxArraySize int) (*ggml.GGML, error) {
 // NewLlamaServer will run a server for the given GPUs
 func NewLlamaServer(gpus discover.GpuInfoList, modelPath string, f *ggml.GGML, adapters, projectors []string, opts api.Options, numParallel int) (LlamaServer, error) {
 	var llamaModel *llama.Model
-	var textProcessor model.TextProcessor
+	var ollamaModel model.Model
 	var err error
 	if envconfig.NewEngine() || f.KV().OllamaEngineRequired() {
 		if len(projectors) == 0 {
-			textProcessor, err = model.NewTextProcessor(modelPath)
+			ollamaModel, err = model.New(modelPath, ml.BackendParams{})
```

Comment:
This is going to trigger the initialization of a lot of GGML code, such as GPU discovery. I'm not sure if we really want to do that here.

In particular, I think there is a memory leak of several GGML data structures, since we don't close the model that is created.